dotnet-client-libraries/HowToUseSftpForSecureFileUploadInDotNet/Test/SshNetSftpClientLiveTests.cs

```diff
@@ -0,0 +1,123 @@
+using Renci.SshNet;
+using Renci.SshNet.Common;
+using System.Net.Sockets;
+using Xunit.Abstractions;
+
+namespace Test;
+
+/**
+ * We can easily set a test server to run these tests against using "Rebex Tiny SFTP Server"
+ * 
+ * https://www.rebex.net/tiny-sftp-server/
+ */
+public class SshNetSftpClientLiveTests
+{
+    private readonly ITestOutputHelper _output;
+
+    public SshNetSftpClientLiveTests(ITestOutputHelper output)
+    {
+        _output = output;
+    }
+
+    [Fact]
+    public void WhenUsingSshNet_ThenCanConnectUsingUsernameAndPassword()
+    {
+        using SftpClient client = new("192.168.1.42", 2222, "tester", "password");
+        client.Connect();
+
+        Assert.True(client.IsConnected);
+
+        client.Disconnect();
+    }
+
+    [Fact]
+    public void WhenUsingSshNet_ThenCanConnectUsingPrivateKey()
+    {
+        var privateKey = new PrivateKeyFile("./ssh-keys/userkey");
+        using SftpClient client = new("192.168.1.42", 2222, "tester", privateKey);
+        client.Connect();
+
+        Assert.True(client.IsConnected);
+
+        client.Disconnect();
+    }
+
+    [Fact]
+    public void WhenUsingSshNet_ThenCanCheckHostFingerprint()
+    {
+
+        var privateKey = new PrivateKeyFile("./ssh-keys/userkey");
+
+        // Change the hardcoded fingerprint to match your test server's.
+        var expectedFingerprint = Convert.FromHexString("D29B9B75F725D9164EA0D4935AFCACD3");
+
+        using SftpClient client = new("192.168.1.42", 2222, "tester", privateKey);
+        client.HostKeyReceived += delegate (object? sender, HostKeyEventArgs e)
+        {
+            e.CanTrust = e.FingerPrint.SequenceEqual(expectedFingerprint);
+        };
```

Comment:
the ``sender`` parameter is not used, can be replaced with a discard operator ``_``
```suggestion
        client.HostKeyReceived += delegate (object? _, HostKeyEventArgs e)
        {
            e.CanTrust = e.FingerPrint.SequenceEqual(expectedFingerprint);
        };
```